app/lib/common/widgets/share/widgets/external_share_options.dart

```diff
@@ -61,59 +48,129 @@ class ExternalShareOptions extends ConsumerWidget {
-                shareToItemUI(
-                  name: lang.qr,
-                  iconWidget: Icon(PhosphorIcons.qrCode()),
-                  color: Colors.grey.shade600,
-                  onTap: onTapQr,
-                ),
-              if (onTapCopy != null)
-                shareToItemUI(
-                  name: lang.copyLink,
-                  iconWidget: Icon(PhosphorIcons.link()),
-                  color: Colors.blueGrey,
-                  onTap: onTapCopy,
-                ),
-              if (onTapSignal != null && isSignalInstalled)
-                shareToItemUI(
-                  name: lang.signal,
-                  iconWidget: Image.asset(
-                    'assets/icon/signal_logo.png',
-                    height: 25,
-                    width: 25,
-                  ),
-                  color: Colors.blue,
-                  onTap: onTapSignal,
-                ),
-              if (onTapWhatsApp != null && isWhatsAppInstalled)
-                shareToItemUI(
-                  name: lang.whatsApp,
-                  iconWidget: Icon(PhosphorIcons.whatsappLogo()),
-                  color: Colors.green,
-                  onTap: onTapWhatsApp,
-                ),
-              if (onTapTelegram != null && isTelegramInstalled)
-                shareToItemUI(
-                  name: lang.telegram,
-                  iconWidget: Icon(PhosphorIcons.telegramLogo()),
-                  color: Colors.blue,
-                  onTap: onTapTelegram,
-                ),
-              if (onTapMore != null)
-                shareToItemUI(
-                  name: lang.more,
-                  iconWidget: Icon(PhosphorIcons.dotsThree()),
-                  color: Colors.grey.shade800,
-                  onTap: onTapMore,
-                ),
+              if (qrContent != null) qrOptionsUI(context, qrContent!),
+              if (shareContentBuilder != null)
+                shareOptionsUI(context, ref, shareContentBuilder!),
             ],
           ),
         ),
       ],
     );
   }
 
+  Widget qrOptionsUI(BuildContext context, String qrContent) {
+    final lang = L10n.of(context);
+    return shareToItemUI(
+      name: lang.qr,
+      iconWidget: Icon(PhosphorIcons.qrCode()),
+      color: Colors.grey.shade600,
+      onTap: () => showQrCode(
+        context,
+        qrContent,
+      ),
+    );
+  }
+
+  Widget shareOptionsUI(
+    BuildContext context,
+    WidgetRef ref,
+    Future<String> Function() shareContentBuilder,
+  ) {
+    final lang = L10n.of(context);
+    final isWhatsAppInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.whatsApp)).valueOrNull ==
+            true;
+    final isWhatsAppBusinessInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.whatsBusiness)).valueOrNull ==
+            true;
+    final isTelegramInstalled =
+        ref.watch(isAppInstalledProvider(ExternalApps.telegram)).valueOrNull ==
+            true;
+    final isSignalInstalled =
```

Comment:
should check that we are on a supported platform - which I think only means `isAndroid` ...